cmake/checks/check_01_cxx_features.cmake

```
@@ -79,6 +81,20 @@ macro(_test_cxx20_support)
     #  error \"insufficient support for C++20\"
     #endif
 
+
+    // Test concepts and requires clauses
+    template <int dim, int spacedim>
+    concept is_valid_dim_spacedim = (dim >= 1 && spacedim <= 3 &&
+                                     dim <= spacedim);
+
+    template <int dim, int spacedim>
+    requires is_valid_dim_spacedim<dim,spacedim>
+    class Triangulation 
+    {};
+
+    Triangulation<1,3> t;
```

```suggestion
    #if !(defined __cpp_concepts) || (__cpp_concepts < 201907L)
    #  error \"insufficient support for C++20\"
    #endif 
```
for conformity (might also be slightly faster).

---

No, I actually would like to keep it as is. If we don't trust that `-std=c++20` works, then we also shouldn't trust that `__cpp_concepts` is set correctly. Let's test for the actual feature we want to use.

---

The point of these `CMake` features check really is that we decide if C++20 is supported based on the capabilities of the compilers with the given flags and not because the user requested C++20 support in any other way (at the moment).  Thus, I wouldn't say that we don't trust that `-std=c++20` works (since compilers allow passing this flag without supporting all of C++20 anyway). On the other hand, I think that the `__cpp_concepts` feature check is reliable, though. Anyway, I rather wanted to suggest using that than requesting changes.